argo/common/run_completion.go

```diff
@@ -20,18 +25,52 @@ var RunCompletionStatuses = struct {
 }
 
 type RunCompletionEvent struct {
-	Status       RunCompletionStatus `json:"status"`
-	PipelineName NamespacedName      `json:"pipelineName"`
+	Status       RunCompletionStatus `json:"status" validate:"required"`
+	PipelineName NamespacedName      `json:"pipelineName" validate:"pipelineName"`
 	// Optionally render structs until https://github.com/golang/go/issues/11939 is addressed
-	RunConfigurationName  *NamespacedName `json:"runConfigurationName,omitempty"`
-	RunName               *NamespacedName `json:"runName,omitempty"`
-	RunId                 string          `json:"runId"`
+	RunConfigurationName  *NamespacedName `json:"runConfigurationName"`
+	RunName               *NamespacedName `json:"runName"`
```

Comment:
These should still be omitempty.

The validation here should be exactly one should be populated (with at least a name (validate both can be a separate card in future)).

if both populated - error
if neither populated - error
if one populated then all good.

Reply:
changed in [f7076ff](https://github.com/sky-uk/kfp-operator/pull/352/commits/f7076ffe6e057088e9725fc23b568083c3e76cab)